### PR TITLE
Feat map

### PR DESCRIPTION
**Pull request by lindatroesken**
Add Map to frontend
Add latitude and longitude to address entity and dto

### CODE REVIEW

File: frontend/src/components/Navbar.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default function Navbar({ user, ...props }) {
       </NavLink>
       {!user && <NavLink to="/login"> Login </NavLink>}
       {user && <NavLink to="/logout"> Logout </NavLink>}
+      {user && <NavLink to="/map/view"> Karte </NavLink>}
       {user && <NavLink to="/profile/view"> Profil </NavLink>}
       {user && <NavLink to="/mangel/new"> Neu </NavLink>}
       {user && <NavLink to="/mangel/list"> Liste </NavLink>}
```
Comment (jamarob) [inline frontend/src/components/Navbar.js:16]:
```suggestion
      {!user && <NavLink to="/login"> Login </NavLink>}
      {user && (<>
        <NavLink to="/logout"> Logout </NavLink>
        <NavLink to="/map/view"> Karte </NavLink>
        <NavLink to="/profile/view"> Profil </NavLink>
        <NavLink to="/mangel/new"> Neu </NavLink>
        <NavLink to="/mangel/list"> Liste </NavLink>
        </>)}
```

File: frontend/src/pages/MangelMap.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import Header from '../components/Header'
+import Page from '../components/Page'
+import Main from '../components/Main'
+import { useEffect, useState } from 'react'
+import ReactMapGL, { Marker, Popup } from 'react-map-gl'
+import { mangelLocations } from '../services/locations-services'
+import styled from 'styled-components/macro'
+
+export default function MangelMap() {
+  const [viewport, setViewport] = useState({
```
Comment (jamarob) [inline frontend/src/pages/MangelMap.js:10]: Extract initial state to a const defined outside of MangelMap

File: frontend/src/pages/MangelMap.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  const [selectedMangel, setSelectedMangel] = useState(null)
+
+  useEffect(() => {
+    const listener = e => {
+      if (e.key === 'Escape') {
+        setSelectedMangel(null)
+      }
+    }
+    window.addEventListener('keydown', listener)
+    return () => {
```
Comment (jamarob) [inline frontend/src/pages/MangelMap.js:27]: Very good!

File: frontend/src/pages/MangelMap.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    }
+  }, [])
+
+  return (
+    <Page>
+      <Header title="Kiezübersicht" />
+      <Main>
+        <ReactMapGL
+          {...viewport}
+          mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
```
Comment (jamarob) [inline frontend/src/pages/MangelMap.js:38]: Read the env at the toplevel of this file

File: frontend/src/pages/MangelMap.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          onViewportChange={viewport => {
+            setViewport(viewport)
+          }}
```
Comment (jamarob) [inline frontend/src/pages/MangelMap.js:41]:
```suggestion
          onViewportChange={setViewport}
```

File: frontend/src/pages/MangelMap.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            <StyledMarker
+              key={mangel.id}
+              longitude={mangel.longitude}
+              latitude={mangel.latitude}
+            >
+              <MarkerButton
+                onClick={e => {
+                  e.preventDefault()
+                  setSelectedMangel(mangel)
+                }}
```
Comment (jamarob) [inline frontend/src/pages/MangelMap.js:54]: Extract to helper function for readability

File: frontend/src/pages/MangelMap.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+              />
+              <div>{mangel.category}</div>
+            </StyledMarker>
+          ))}
+          {selectedMangel && (
+            <StyledPopup
+              longitude={selectedMangel.longitude}
+              latitude={selectedMangel.latitude}
+              onClose={() => setSelectedMangel(null)}
```
Comment (jamarob) [inline frontend/src/pages/MangelMap.js:63]: see above maybe name it `clearSelected`?

File: frontend/src/services/locations-services.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+export const mangelLocations = [
```
Comment (jamarob) [inline frontend/src/services/locations-services.js:1]: This is just some mock data, that is going to be removed?

**Review by jamarob — APPROVED** [review]
Nice! 👍 Easy to review. Left some comments, but nothing critical.

File: frontend/src/services/locations-services.js (hunk 8 above)
Comment (lindatroesken) [reply frontend/src/services/locations-services.js:1]: yes, just temporary, until I get data from backend

File: frontend/src/pages/MangelMap.js (hunk 7 above)
Comment (lindatroesken) [reply frontend/src/pages/MangelMap.js:63]: 👍

File: frontend/src/pages/MangelMap.js (hunk 6 above)
Comment (lindatroesken) [reply frontend/src/pages/MangelMap.js:54]: 👍

**Pull request merged**